report/renderer/report-ui-features.js

```diff
@@ -319,25 +320,29 @@ export class ReportUIFeatures {
    * @return {Array<HTMLElement>}
    */
   _getThirdPartyRows(rowEls, finalDisplayedUrl) {
-    /** @type {Array<HTMLElement>} */
-    const thirdPartyRows = [];
     const finalDisplayedUrlRootDomain = Util.getRootDomain(finalDisplayedUrl);
+    const mainEntityName = this.json.entities?.firstParty;
 
+    /** @type {Array<HTMLElement>} */
+    const thirdPartyRowEls = [];
     for (const rowEl of rowEls) {
-      if (rowEl.classList.contains('lh-sub-item-row')) continue;
```

Comment:
Was removing this intentional?

Reply:
I think I've introduced a regression in third-party entity count shown on the filter, with that line removed. We don't have a unit test to check if the count displayed is correct, so I'm going to add one for that as well.

Longer version:

The legacy filter checkbox visibility works by matching rows that have a `lh-text__url` and a `dataset.url` attached to it. The filter checkbox is hidden if `count(such rows) === count(all table rows)` in which case all rows are third-parties, or if `count(such rows)` equals to zero, in which case it assumes all rows are 1st parties. 

For an audit like `third-party-summary`, this still works when the third-party count mismatches and fails the all-third-party check. The 1st party check (incorrectly) succeeds. I think we can rewrite this section with both comparisons basing on primary rows only.

Reply:
Refactored the code; I think it should be good for a re-review.